salt/states/boto3_route53.py

```diff
@@ -798,3 +826,204 @@ def rr_absent(name, HostedZoneId=None, DomainName=None, PrivateZone=False,
         ret['result'] = False
 
     return ret
+
+def health_check_present(name, Name=None, CallerReference=None, IPAddress=None,
+                         Port=None, Type=None, ResourcePath=None, FullyQualifiedDomainName=None, SearchString=None,
+                         RequestInterval=None, FailureThreshold=None, MeasureLatency=None, Inverted=None, Disabled=None,
+                         HealthThreshold=None, ChildHealthChecks=None, EnableSNI=None, Regions=None, AlarmIdentifier=None,
+                         InsufficientDataHealthStatus=None, region=None, key=None, keyid=None, profile=None):
+    '''
+    Ensure the Route53 health check is present.
+
+    name
+        The name of the state definition.  This will be used for Name if the latter is
+        not provided.
+
+    Name
+        Name of the health check.
+
+    region
+        The region to connect to.
+
+    key
+        Secret key to be used.
+
+    keyid
+        Access key to be used.
+
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    '''
+
+    Name = Name if Name else name
+
+    if Type is None:
+        raise SaltInvocationError("'Type' is a required parameter when adding or updating resource records.")
+
+    ret = {'name': name, 'result': True, 'comment': '', 'changes': {}}
+    params = locals()
+    create = False
+    update = False
+    reset_elements = []
+    updatables = [
+        'HealthCheckVersion', 'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+        'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+        'EnableSNI', 'InsufficientDataHealthStatus'
+    ]
+    resettables = [ 'FullyQualifiedDomainName', 'Regions', 'ResourcePath', 'ChildHealthChecks' ]
+
+    checks = __salt__['boto3_route53.find_health_checks'](Name=Name, region=region, key=key, keyid=keyid, profile=profile)
+
+    if len(checks) > 1:
+        ret['comment'] = 'Given criteria matched more than one health check.'
+        log.error(ret['comment'])
+        ret['result'] = False
+        return ret
+    elif len(checks) == 0:
+        create = True
+        CallerReference = CallerReference if CallerReference else str(uuid.uuid4())  # future lint: disable=blacklisted-function
+    else:
+        check = checks[0].get('HealthCheckConfig')
+        for u in updatables:
+            if u in [ 'HealthCheckVersion' ]:
+                continue
+            if _ensure_unicode(params.get(u)) != _ensure_unicode(check.get(u)):
+                update = True
+                break
+            if ((ChildHealthChecks is not None) and (check.get('ChildHealthChecks') is not None)):
```

Comment:
If `''` is not valid or irrelevant here, then I would rewrite this as
```suggestion
            if ChildHealthChecks and check.get('ChildHealthChecks'):
````

I'd do the same with the other checks below here

Reply:
This code is quite tricky, as not all values are strings, nor should a 0 be treated like a none.